fetch_crypto_prices.py

```diff
@@ -2,24 +2,69 @@
 import json
 import os
 from datetime import datetime
+import logging
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
 
 # Get BASE_URL from environment variable
 BASE_URL = os.getenv('BASE_URL')
 
-def fetch_crypto_prices():
-    url = f"{BASE_URL}/simple/price?ids=dai,ethereum,bitcoin&vs_currencies=usd"
-    response = requests.get(url)
-    if response.status_code == 200:
+# Setup retry strategy for network issues
+retry_strategy = Retry(
+    total=3,  # Retry 3 times
+    backoff_factor=1,  # Wait between retries (exponential backoff)
+    status_forcelist=[429, 500, 502, 503, 504],  # Retry on these HTTP status codes
+    method_whitelist=["GET"]
+)
+
+# Attach the retry strategy to a session
+adapter = HTTPAdapter(max_retries=retry_strategy)
+session = requests.Session()
+session.mount("https://", adapter)
+session.mount("http://", adapter)
+
+def fetch_crypto_prices(crypto_ids=["dai", "ethereum", "bitcoin"], currency="usd"):
+    """
+    Fetch the latest crypto prices for the given crypto_ids.
+    """
+    if BASE_URL is None:
+        logging.error("BASE_URL is not set in environment variables")
+        return False
+    
+    url = f"{BASE_URL}/simple/price?ids={','.join(crypto_ids)}&vs_currencies={currency}"
+    
+    try:
+        logging.info(f"Requesting crypto prices from {url}")
+        response =_for_status()  # Raises exception for 4xx/5xx responses
         data = response.json()
-        # Save the data to a JSON file
-        with open('crypto_prices.json', 'w') as f:
+
+        # Validate response
+        if not all(crypto in data for crypto in crypto_ids):
+            logging.error(f"Response is missing expected keys: {crypto_ids}")
+            return False
+        
+        # Save the data with a timestamp
+        timestamp = datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+        filename = f'crypto_prices_{timestamp}.json'
```

Comment:
How will our [React components](https://github.com/fantomstarterio/futurestarter-react-embed) be able to [fetch the price data](https://github.com/fantomstarterio/crypto-prices-cache?tab=readme-ov-file#usage) if we suddenly have a dynamic filename? Or are we not going to do that anymore? If not, then this whole repository will be useless right?